### PR TITLE
jetbrains: Fix update.pl script for MPS and update

### PR DESCRIPTION
**Pull request by avdv**
###### Motivation for this change

New upstream releases.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132043` at 544e20e2 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/f27704af57443c3dc347775445627a28">jetbrains.clion</a></li>
    <li><a href="https://gist.github.com/7bdef6a71978b5932170e7e07095abb6">jetbrains.datagrip</a></li>
    <li><a href="https://gist.github.com/f3ab64d4bfb7741f1d7b91ba8c924956">jetbrains.goland</a></li>
    <li><a href="https://gist.github.com/95ee54997a77dd10c06f85fdbe7aec85">jetbrains.mps</a></li>
    <li><a href="https://gist.github.com/cab101be95e5ab4f63b42efed9b86636">jetbrains.phpstorm</a></li>
    <li><a href="https://gist.github.com/fca1630f7d17a12b42d4ab7fe7279e7e">jetbrains.pycharm-community</a></li>
    <li><a href="https://gist.github.com/3e2ca76b38711b429f394ea387608dc2">jetbrains.pycharm-professional</a></li>
    <li><a href="https://gist.github.com/535b8973aa8e8cf64bae49a580b0808f">jetbrains.rider</a></li>
    <li><a href="https://gist.github.com/8e80135fcff1e56c0bd57bc5217c8a47">jetbrains.ruby-mine</a></li>
    <li><a href="https://gist.github.com/9aa70ed78f394d269d935b4c148a5b46">jetbrains.webstorm</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by avdv** [comment]
AFAICS, these packages are x86_64-linux only. They also provide dmg packages for MacOS, but I don't know how we could deal with them.

Should we constrain the build to x86_64-linux here?

**Comment by avdv** [comment]
Seems like they have changed the packaging, all the newer versions fail for some reason, such as:
```
patchelf: getting info about 'bin/fsnotifier64': No such file or directory
```
I'll look into this...

File: pkgs/applications/editors/jetbrains/common.nix [hunk 1]
```diff
@@ -52,7 +52,7 @@ with stdenv; lib.makeOverridable mkDerivation rec {
       }
 
       interpreter=$(echo ${stdenv.glibc.out}/lib/ld-linux*.so.2)
-      if [ "${stdenv.hostPlatform.system}" == "x86_64-linux" ]; then
+      if [ "${stdenv.hostPlatform.system}" == "x86_64-linux" -a -e bin/fsnotifier64 ]; then
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/jetbrains/common.nix:55]:
```suggestion
      if [[ "${stdenv.hostPlatform.system}" == "x86_64-linux" && -e bin/fsnotifier64 ]]; then
```

Comment (avdv) [reply pkgs/applications/editors/jetbrains/common.nix]: applied.

**Comment by avdv** [comment]
> AFAICS, these packages are x86_64-linux only. They also provide dmg packages for MacOS, but I don't know how we could deal with them.
> 
> Should we constrain the build to x86_64-linux here?

Specifically, the build for jetbrains.idea-community.x86_64-darwin was last successful on 2020-03-14 08:14:10, which was for idea-community 2019.3.3 (https://hydra.nixos.org/build/148157165)

**Comment by SebTM** [comment]
Tested with NixOS-Unstable in PHP/Bash-Environment - changes LGTM and working were working :)

**Comment by fabianhjr** [comment]
Built and tested idea-community. LGTM

**Comment by Ma27** [comment]
@avdv would you mind opening a backport PR manually? :)

**Comment by avdv** [comment]
> @avdv would you mind opening a backport PR manually? :)

Sure thing! Done: #133025

**Pull request merged**